### PR TITLE
Enum members now also link to the corresponding class in the webdocs

### LINKED ISSUE #13254: Expose enums in documentation in a way that allows hyperlinks
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Any OS, current master branch (f5c513ca7)

**Issue description:**
We have support for keeping track of enums in the classref's XML, with this form:
```
Light2D.xml
28:             <member name="mode" type="int" setter="set_mode" getter="get_mode" enum="Light2D.Mode">
61:             <member name="shadow_filter" type="int" setter="set_shadow_filter" getter="get_shadow_filter" enum="Light2D.ShadowFilter">
```

But currently there's no real link done with the actual constants from that enum:
```
Light2D.xml
29:                     The Light2D's mode. See MODE_* constants for values.
81:             <constant name="MODE_ADD" value="0">
84:             <constant name="MODE_SUB" value="1">
87:             <constant name="MODE_MIX" value="2">
90:             <constant name="MODE_MASK" value="3">
```

We should add a way to properly show and reference enums in the oflfine and online docs, so that doc writers can write `[enum Light2D.Mode]` to create a reference to the enum section where the relevant constants are defined.

### PR DESCRIPTION
**Pull request by fodinabor**
Fixes #13254

### CODE REVIEW

**Comment by akien-mga** [comment]
Nice work, thanks!

I think we should go further than just linking to the class, and properly display the enums in a new "Enumerations" section as done in the editor: https://github.com/godotengine/godot/issues/13254#issuecomment-346888784
Then each enum could have a reference in rst, which could be used to create hyperlinks when parsing the `[enum EnumName]` stuff.

File: doc/tools/makerst.py [hunk 1]
```diff
@@ -276,6 +276,15 @@ def make_type(t):
         return ':ref:`' + t + '<class_' + t.lower() + '>`'
     return t
 
+def make_enum(t, f):
```
Comment (akien-mga) [inline doc/tools/makerst.py]: The `f` parameter is unused?

**Comment by fodinabor** [comment]
image


got you some enums here

**Comment by akien-mga** [comment]
Awesome work! I'll review that tomorrow.

**Pull request merged**